
leg:
  leg_acronyms: ["LF", "RF", "LB", "RB"] # left front, right front, left back, right back
  joint_names: ["LF_theta1", "LF_theta2", "LF_theta3",
                "RF_theta1", "RF_theta2", "RF_theta3",
                "LB_theta1", "LB_theta2", "LB_theta3",
                "RB_theta1", "RB_theta2", "RB_theta3"]
  link1_length: 5 # hip [cm]
  link2_length: 10 # thigh [cm]
  link3_length: 9 # shin [cm]

body:
  width: 10 # [cm]
  length: 24 # [cm]
  height: 6 # [cm]

walking:
  deg_per_sec: 120
  max_step_height_pct: 0.2
  step_length_deg: 20 # with leg fully straight, degree rotated forward for max step length
  LF_phase_offset: 0 # left front leg phase offset in degrees
  RF_phase_offset: 180 # right front leg phase offset in degrees
  LB_phase_offset: 180 # left back leg phase offset in degrees
  RB_phase_offset: 0 # right back leg phase offset in degrees

control:
  control_hz: 50 # control loop frequency in Hz

neutral_stance:
  neutral_stance_height_pct: 0.7 # percentage of straight leg height

servos:
  servo_map:
    LF_theta1: 0 
    LF_theta2: 1 
    LF_theta3: 2 
    RF_theta1: 3
    RF_theta2: 4
    RF_theta3: 5
    LB_theta1: 6
    LB_theta2: 7
    LB_theta3: 8
    RB_theta1: 9
    RB_theta2: 10
    RB_theta3: 11
  
  is_arm_rotate: 
    # used as multiplier; 
    # 1 if servo arm rotates the joint
    # -1 if servo body rotates the joint
    LF_theta1: 1
    LF_theta2: -1 
    LF_theta3: -1 
    RF_theta1: 1
    RF_theta2: -1
    RF_theta3: -1
    LB_theta1: 1
    LB_theta2: -1
    LB_theta3: -1
    RB_theta1: 1
    RB_theta2: -1
    RB_theta3: -1

  is_actuation_axis_co_rotate:
    # used as multiplier; 
    # 1 if servo actuation positive rotation direction matches joint Z-axis   
    # -1 if not
    LF_theta1: 1
    LF_theta2: -1
    LF_theta3: -1
    RF_theta1: 1
    RF_theta2: -1
    RF_theta3: -1
    LB_theta1: -1
    LB_theta2: -1
    LB_theta3: -1
    RB_theta1: -1
    RB_theta2: -1
    RB_theta3: -1


  offsets:
    LF_theta1: 0 
    LF_theta2: 0 
    LF_theta3: 0 
    RF_theta1: 0
    RF_theta2: 0
    RF_theta3: 0
    LB_theta1: 0
    LB_theta2: 0
    LB_theta3: 0
    RB_theta1: 0
    RB_theta2: 0
    RB_theta3: 0

  min_angles:
    LF_theta1: 30 
    LF_theta2: 30
    LF_theta3: 90
    RF_theta1: 30
    RF_theta2: 30
    RF_theta3: 90
    LB_theta1: 30
    LB_theta2: 30
    LB_theta3: 90
    RB_theta1: 30
    RB_theta2: 30
    RB_theta3: 90

  max_angles:
    LF_theta1: 150
    LF_theta2: 150
    LF_theta3: 180
    RF_theta1: 150
    RF_theta2: 150
    RF_theta3: 180
    LB_theta1: 150
    LB_theta2: 150
    LB_theta3: 180
    RB_theta1: 150
    RB_theta2: 150
    RB_theta3: 180

  actuation_range:
    LF_theta1: 270
    LF_theta2: 270
    LF_theta3: 270
    RF_theta1: 270
    RF_theta2: 270
    RF_theta3: 270
    LB_theta1: 270
    LB_theta2: 270
    LB_theta3: 270
    RB_theta1: 270
    RB_theta2: 270
    RB_theta3: 270

  pulsewidth_range:
    LF_theta1: [500, 2500]
    LF_theta2: [500, 2500]
    LF_theta2: [500, 2500]
    LF_theta3: [500, 2500]
    RF_theta1: [500, 2500]
    RF_theta2: [500, 2500]
    RF_theta3: [500, 2500]
    LB_theta1: [500, 2500]
    LB_theta2: [500, 2500]
    LB_theta3: [500, 2500]
    RB_theta1: [500, 2500]
    RB_theta2: [500, 2500]
    RB_theta3: [500, 2500]

visualization:
  joint_limits:
    hip_lower: -90
    hip_upper: 90
    thigh_lower: -90
    thigh_upper: 90
    shin_lower: -90
    shin_upper: 90